packages/commonwealth/client/scripts/models/Account.ts

```diff
@@ -95,6 +108,8 @@ class Account {
       jwt: app.user.jwt,
       signature,
       wallet_id: this.walletId,
+      session_public_address: app.sessions.getAddress(this.chain.node.ethChainId || 1),
```

Comment:
do you know how could this be generalised to work for other chains? @raykyri

Reply:
The sessions controller should generate signers for chains of different types!

Reply:
It looks like depending on the network (ethereum/substrate/cosmos) that is being used, the current chain id (e.g. 1, 5, osmosis, edgeware) is stored in different values... What is the difference between `this.chain.id`, `this.chain.node.id` and `this.chain.node.ethChainId`? Ideally at this point in the code we would have a generic way of getting the current chain id, otherwise I'm thinking we could move that code into the various wallet classes...